src/main/kotlin/com/goodchoice/infra/email/service/EmailTemplateService.kt

```diff
@@ -0,0 +1,29 @@
+package com.goodchoice.infra.email.service
+
+import com.goodchoice.infra.email.model.EmailMessageContent
+import com.goodchoice.infra.email.model.EmailTemplateInput
+import com.goodchoice.infra.email.model.EmailTemplateInput.EmployeeInvitation
+import org.thymeleaf.context.Context
+import org.thymeleaf.ITemplateEngine as TemplateEngine
+
+interface EmailTemplateService {
+
+    fun render(input: EmailTemplateInput): EmailMessageContent
+}
+
+class ThymeleafEmailTemplateService(private val templates: TemplateEngine) : EmailTemplateService {
+
+    override fun render(input: EmailTemplateInput): EmailMessageContent =
+        when (input) {
+            is EmployeeInvitation -> EmailMessageContent(
+                subject = "Employee invitation",
+                body = renderBody("email/employee-invitation", input)
+            )
+        }
+
+    @Suppress("SameParameterValue")
```

Comment:
What does that do?

Reply:
It suppresses warning `Actual value of parameter 'templateName' is always 'email/employee-invitation' ` for method. We should suppress warnings that are not relevant. It will be removed after another template adding